hil-test/src/bin/multicore.rs

```diff
@@ -0,0 +1,167 @@
+//! Multicore flash cache coherency test.
+//!
+//! This test verifies that flash operations remain correct when running
+//! on one core while the other core creates significant cache pressure.
+//! It tests the `multicore_auto_park()` functionality. The `multicore_ignore` is not easily tested
+//! with HIL, so it is present in the `qa-test` instead.
+
+//% CHIPS: esp32 esp32s3
+//% FEATURES: unstable esp-storage esp-storage/defmt defmt
+
+#![no_std]
+#![no_main]
+#![feature(asm_experimental_arch)]
+
+use core::ptr::addr_of_mut;
+
+use defmt::info;
+use embedded_storage::nor_flash::NorFlash;
+use esp_hal::{
+    clock::CpuClock,
+    delay::Delay,
+    peripherals::FLASH,
+    system::{CpuControl, Stack},
+    time::{Duration, Instant},
+};
+use esp_storage::FlashStorage;
+use hil_test as _;
+
+// Pick which core does flash access
+const FLASH_ON_CORE_0: bool = true;
+
+// Test write location (must be in writable flash region)
+const TEST_ADDR: u32 = 0x9000;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "esp32")] {
+        const CACHE_PRESSURE_BASE: u32 = 0x400C_2000;
+    }
+    else {
+        const CACHE_PRESSURE_BASE: u32 = 0x4200_0000;
+    }
+}
```

Comment:
```suggestion
const CACHE_PRESSURE_BASE: u32 = if cfg!(feature = "esp32") {
    0x400C_2000
} else {
    0x4200_0000
};
```

It doesn't _have to_ be cfg-if all the time